tests/utils/test_ellipsoid_utils.py

```diff
@@ -13,6 +16,88 @@
 class TestEllipsoidUtils(unittest.TestCase):
     """Class containing all unit tests for ellipsoid utils."""
 
+    def test_get_ortho_axis_alignment_transform(self) -> None:
+        """Tests the get_otho_axis_alignment_transform() function with a GtsfmData object containing 3 camera frustums
+        and 6 points in the point cloud. All points lie on z=0 plane. All frustums lie on z=2 plane and look down on
+        the z=0 plane.
+
+           sample_data:              output_data:
+
+               y                         y
+               |                         |
+           o   |                         o
+             c | c                       |
+          ------------- x   ==>  --o--c-----c--o-- x
+             o | c                       |
+               |   o                     c
+               |                         |
+
+        c = point at (xi,yi,0) with a camera frustum at (xi,yi,2)
+        o = point at (xi,yi,0)
+        """
+
+        sample_data = GtsfmData(number_images=3)
+        default_intrinsics = Cal3Bundler(fx=100, k1=0, k2=0, u0=0, v0=0)
+
+        # Add 3 camera frustums to sample_data (looking down at z=0 plane)
+        camera0 = PinholeCameraCal3Bundler(Pose3(Rot3(np.eye(3)), np.array([-1, 1, 2])), default_intrinsics)
+        camera1 = PinholeCameraCal3Bundler(Pose3(Rot3(np.eye(3)), np.array([1, 1, 2])), default_intrinsics)
+        camera2 = PinholeCameraCal3Bundler(Pose3(Rot3(np.eye(3)), np.array([1, -1, 2])), default_intrinsics)
+
+        sample_data.add_camera(0, camera0)
+        sample_data.add_camera(1, camera1)
+        sample_data.add_camera(2, camera2)
+
+        # Add 6 tracks to sample_data
+
+        # fmt: off
+        points3d = np.array(
+            [
+                [1, 1, 0],
+                [-1, 1, 0],
+                [-2, 2, 0],
+                [-1, -1, 0],
+                [1, -1, 0],
+                [2, -2, 0],
+                [5, 5, 0]
```

Comment:
added this last point as an outlier point to be filtered out inside the ellipsoid utils wrapper function